### PR TITLE
[Dynamo] Treat integers stored on nn.Modules as dynamic

### LINKED ISSUE #115711: [torch.compile] nn.Module int/float attributes caused re-compile
### 🚀 The feature, motivation and pitch

Repro:
```
import torch

class MyModule(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(4, 4)
        self.step = 0

    def forward(self, x: torch.Tensor) -> torch.Tensor:
        x = x + 1
        self.step += 1
        return self.layer(x)

m = MyModule()
opt_m = torch.compile(backend="eager")(m)

x = torch.randn(3, 4)
print(opt_m(x))

x = torch.randn(4, 4)
print(opt_m(x))

x = torch.randn(5, 4)
print(opt_m(x))
```

In this example, ```self.step``` is a constant attribute of ```nn.Module```. Though it's not part of FX graph, Dynamo would produce guard for this and triggers re-compilation since this is used as a counter as model iteration goes.
Probably we can wrap the int/float attributes of ```nn.Module``` as ```torch.SymInt/Float``` to avoid guard failures and re-compilations, but only do this when they are not used in control flow.

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @jansel 

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes #115711

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py [hunk 1]
```diff
@@ -472,8 +472,9 @@ def generate_example_inputs(batch: int, seq: int, hidden_size: int):
 
         if test_local_backend:
             assert local_ort is not None
-            number_of_captured_graphs = 2 if test_backward else 1
+            number_of_captured_graphs = 3 if test_backward else 2
             execution_count = len(example_args_collection) * number_of_captured_graphs
+            # breakpoint()
```
Comment (BowenBao) [inline test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py:477]: Nit: remove debug code

**Pull request closed without merging**